alica_engine/src/engine/modelmanagement/ModelManager.cpp

```diff
@@ -36,13 +40,22 @@ ModelManager::ModelManager(ConfigChangeListener& configChangeListener, const std
     Factory::setModelManager(this);
 }
 
-Plan* ModelManager::loadPlanTree(const std::string& masterPlanName)
+Plan* ModelManager::loadPlanTree(const std::string& masterPlanName, std::optional<std::string> placeholderMapping)
```

Comment:
IMO it's better if we left the file loading to the application and took in a loaded file as a string/etc that can be parsed to yaml.  That way we can potentially even read the full file from e.g. the ROS param server (allowing fussing with placeholders using rosarg).  Requiring the mapping to be written in hardcoded file disallows some customization

Reply:
sure, can be done

Reply:
I implemented this